src/matrixvariate.jl

```diff
@@ -0,0 +1,218 @@
+# Adapted from Distributions.jl
+
+## Wishart
+
+using StatsFuns: logtwo, logmvgamma
+
+struct TuringWishart{T<:Real, ST} <: ContinuousMatrixDistribution
+    df::T     # degree of freedom
+    chol::ST  # the Cholesky of scale matrix
+    c0::T     # the logarithm of normalizing constant in pdf
+end
+
+#### Constructors
+
+function TuringWishart(df::T, S::AbstractMatrix{T}) where {T <: Real}
+    p = size(S, 1)
+    df > p - 1 || error("dpf should be greater than dim - 1.")
+    C = cholesky(S)
+    return TuringWishart(df, C)
+end
+function TuringWishart(df::T, C::Cholesky{T}) where {T <: Real}
+    c0 = _wishart_c0(df, C)
+    R = Base.promote_eltype(T, c0)
+    return TuringWishart(R(df), C, R(c0))
+end
+
+function TuringWishart(df::Real, S::AbstractMatrix)
+    T = Base.promote_eltype(df, S)
+    TuringWishart(T(df), convert(AbstractArray{T}, S))
+end
+
+function _wishart_c0(df::Real, C::Cholesky)
+    h_df = df / 2
+    p = size(C, 1)
+    h_df * (logdet(C) + p * typeof(df)(logtwo)) + logmvgamma(p, h_df)
+end
+
+#### Properties
+
+Distributions.insupport(::Type{TuringWishart}, X::Matrix) = isposdef(X)
+Distributions.insupport(d::TuringWishart, X::Matrix) = size(X) == size(d) && isposdef(X)
+
+PDMats.dim(d::TuringWishart) = size(d.chol, 1)
```

Comment:
That should not be needed? The interface in Distributions only uses `size` and `length` (and with https://github.com/TuringLang/Turing.jl/pull/1076 it's not used in Turing anymore).
```suggestion
```

Reply:
It's not needed outside but it's used in the same file in many places.

Reply:
Yes but one should really just use `length` there.

Reply:
Doesn't have to be `PDMats.dim` though, can be just a local `dim` function. `length` can also be defined.

Reply:
Sure, that would be fine. It just feels weird to define a function that does not seem to be part of the official Distributions interface and is taken from PDMats, even though we do not work with `AbstractPDMat`s.